modules/application/application.js

```diff
@@ -0,0 +1,170 @@
+      check.addEventListener('click', (event) => {
+        const refObj = event.currentTarget;
+        if (refObj.ref.todoList[refObj.index].completed) {
+          refObj.ref.todoList[refObj.index].completed = false;
+        } else {
+          refObj.ref.todoList[refObj.index].completed = true;
+        }
+        refObj.ref.onSaveList();
+        ref.updateDom();
+      });
+      check.index = index;
+      check.ref = ref;
+    }, ref);
+
+    // clear all completed
+    const clearBtn = document.querySelector('.custom-btn');
+    clearBtn.addEventListener('click', (event) => {
+      const refObj = event.currentTarget;
+      const filteredTasks = refObj.ref.todoList.filter((item) => {
+        const state = item.completed === false;
+        return state;
+      });
+      refObj.ref.todoList = filteredTasks;
+      refObj.ref.onSaveList();
+      ref.updateDom();
+    });
+    clearBtn.ref = this;
+  }
+
+  onsubmiteventDispatcher = () => {
+    this.addTask.addEventListener('keyup', (event) => {
+      if (event.keyCode !== 13) {
+        return;
+      }
+
+      const refObj = event.currentTarget;
+      const input = refObj.ref.addTask.value;
+      if (!input.replace(/\s/g, '').length || input.length <= 0) {
+        return;
+      }
+
+      if (event.keyCode === 13) {
+        refObj.ref.todoList.push({
+          index: (refObj.ref.todoList.length + 1), description: input, completed: false,
+        });
+
+        refObj.ref.onSaveList();
+        refObj.ref.addTask.value = '';
+      }
+      refObj.ref.updateDom();
+      event.preventDefault();
+    });
+    this.addTask.ref = this;
+  }
+
+  onediteventDispatcher = () => {
+    const ref = this;
+    const listDesc = document.querySelectorAll('.desc');
+    listDesc.forEach((desc, index) => {
+      desc.addEventListener('keyup', (event) => {
+        if (event.keyCode !== 13) {
+          return;
+        }
+
+        const refObj = event.currentTarget;
+        let input = refObj.value.innerHTML;
+        refObj.value.innerHTML = input.replace('<br>', '');
+        input = refObj.value.innerHTML;
+        if (!input.replace(/\s/g, '').length || input.length <= 0) {
+          return;
+        }
+
+        if (refObj.value.innerHTML !== refObj.ref.todoList[refObj.index].description) {
+          refObj.value.innerHTML = input.replace('<br>', '');
+          refObj.ref.todoList[refObj.index].description = refObj.value.innerHTML;
+          refObj.ref.onSaveList();
+          refObj.value.blur();
+        }
+      });
+      desc.index = index;
+      desc.ref = ref;
+      desc.value = desc;
+    }, ref);
+  }
+
+  onSaveList = () => {
+    localStorage.setItem('application_config', JSON.stringify(this.todoList));
+  }
+
+  onLoadList = () => {
+    if (localStorage.getItem('application_config') != null) {
+      this.todoList = JSON.parse(localStorage.getItem('application_config'));
+    }
+  }
+}
+
+/* eslint-disable */
+export { Application };
```

Comment:
- [ ] **Required**: You should not disable the linters, they provide a standard for team/community code here at Microverse and in your future workplaces. Kindly fix the errors the point out.

Reply:
I wanted to take that out. the linters test would have still passed even if I didn't disable it. I just completely forgot to remove it. 😄

Reply:
Ohh okay, that's nice. Don't forget again to remove it :wink: